File: src/main/java/gestorAplicacion/ventas/VentaIndividual.java
package gestorAplicacion.ventas;

import gestorAplicacion.dominio.Cliente;

import java.io.Serial;
import java.io.Serializable;
import java.util.List;

/**
 * Clase que representa una venta individual en el sistema.
 *
 * @author Juan Daniel Bula, Leidy Daniela Alzate, Oscar Andres Usuga
 */
public class VentaIndividual implements Serializable {

    @Serial
    private static final long serialVersionUID = 4L;

    private String codigoVenta;
    private Cliente cliente;
    private List<Producto> productos;
    private String estado;

    public VentaIndividual(String codigo, Cliente cliente, List<Producto> productos) {
        this.codigoVenta = codigo;
        this.cliente = cliente;
        this.productos = productos;
        this.estado = "PENDIENTE";
    }

    public void venderProductos() {
        this.estado = "FINALIZADA";
        this.productos.forEach(Producto::venderProducto);
    }

    public String getEstado() {
        return estado;
    }

    public void setEstado(String estado) {
        this.estado = estado;
    }

    public String getCodigoVenta() {
        return codigoVenta;
    }

    public List<Producto> getProductos() {
        return productos;
    }

    @Override
    public String toString() {
        return String.format("""
                Venta [
                    codigo = %s,
                    cliente = %s,
                    productos = %s,
                    estado = %s
                ]""", codigoVenta, cliente, productos, estado);
    }
}

File: python/gestorAplicacion/ventas/venta_individual.py
class VentaIndividual:

    def __init__(self, codigo_venta, cliente, productos):
        self._estado = "PENDIENTE"
        self._codigo_venta = codigo_venta
        self._ciente = cliente
        self._productos = productos

    def venderProductos(self):
        self._estado = "FINALIZADA"
        for producto in self._productos:
            producto.venderProducto(producto)

    def devolverProducto(self, codigo_producto):
        for producto in self._productos:
            if codigo_producto == producto.getCodigo():
                self._productos.remove(producto)

        if self._productos is None:
            self.setEstado("DEVUELTA")

    def getEstado(self):
        return self._estado

    def setEstado(self, estado):
        self._estado = estado

    def getCodigoVenta(self):
        return self._codigo_venta

    def getProductos(self):
        return self._productos

    def __str__(self):
        return """"Venta [
        codigo = %s,
        cliente = %s
        productos = %s
        estado = %s
        ]""" % (self._codigo_venta, self._ciente, self._productos, self._estado)
